products/BellHybrid/services/db/agents/QuotesAgent.cpp

```diff
@@ -1,14 +1,15 @@
-// Copyright (c) 2017-2024, Mudita Sp. z.o.o. All rights reserved.
+// Copyright (c) 2017-2025, Mudita Sp. z.o.o. All rights reserved.
 // For licensing, see https://github.com/mudita/MuditaOS/blob/master/LICENSE.md
 
 #include "QuotesAgent.hpp"
 #include "QuotesQueries.hpp"
 #include <Common/Query.hpp>
+#include <service-db/agents/settings/SystemSettings.hpp>
 
 namespace Quotes
 {
-    QuotesAgent::QuotesAgent(Database *quotesDB, std::unique_ptr<settings::Settings> settings)
-        : quotesDB(quotesDB), shuffleQuoteModel(std::move(settings), quotesDB)
+    QuotesAgent::QuotesAgent(Database *quotesDB, std::shared_ptr<settings::Settings> settings)
```

Comment:
I'd try to avoid using shared pointer here. IMO there's no need for `ShuffleQuoteModel` to have ownership of `Settings`, the ownership should be held by `QuotesAgent` and `ShuffleQuoteModel` should only get reference or raw pointer to this object (as it is just a class member, not any standalone object):

 ```suggestion
    QuotesAgent::QuotesAgent(Database *quotesDB, std::unique_ptr<settings::Settings> settings)
    : settings(std::move(settings)), quotesDB(quotesDB), shuffleQuoteModel(*this->settings, quotesDB)
```

ofc that will also require to slightly modify `ShuffileQuoteModel`'s ctor (it should take `Settings &`)

Reply:
done